pkg/controllers/vdb/verticadb_controller.go

```diff
@@ -180,6 +180,8 @@ func (r *VerticaDBReconciler) constructActors(log logr.Logger, vdb *vapi.Vertica
 		// Handle setting up the pod security context. This picks the
 		// UID/fsGroup that we will run with.
 		MakePodSecurityReconciler(r, log, vdb),
+		// Handle wrong images for the operator
+		MakeImageCheckReconciler(r, log, vdb, prunner, pfacts),
```

Comment:
I'm not sure if this is the correct spot. For create/revive, we haven't even created the pods yet. But we also need to do the check early for upgrade code paths. So, we probably need it in a couple of places. I think we need to do the check in the same spots that we capture the version. I would suggest rather than add a new reconciler, lets add your logic to the version reconciler (`version_reconciler.go). Then we checking it in the correct spot.

Reply:
Then should we call it `ImageVersionReconciler`? where we check both the version and the deployment type.